goss-testing/tests/ncn/goss-switch-ping-status.yaml

```
@@ -0,0 +1,43 @@
+#
+# MIT License
+#
+# (C) Copyright 2022 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+{{ $scripts := .Env.GOSS_BASE | printf "%s/scripts" }}
+{{ $logrun := $scripts | printf "%s/log_run.sh" }}
+command:
+    {{ $testlabel := "switch-ping-status" }}
+    {{$testlabel}}:
+        title: Verify that the network switch has external access
+        meta:
+            desc: If this test fails, then validate the network switch configurations.
+            sev: 0
+        exec: |-
+            #Set pipefail so that if any commands in the command pipeline fail, the test will fail
+            set -eo pipefail
+            "{{$logrun}}" -l "{{$testlabel}}" \
```

If you are going to use logrun here, you should avoid having the switch password show up in the log by using the --no-args flag:
```suggestion
            "{{$logrun}}" --no-args -l "{{$testlabel}}" \
```